### PR TITLE
Add versioning note and prepare v1.0.1

### PR DESCRIPTION
**Pull request by adityasaky**
https://github.com/secure-systems-lab/dsse/pull/61#discussion_r1596858848

### CODE REVIEW

**Review by MarkLodato — APPROVED** [review]
It would be good to get a few more maintainers to approve, just to make sure we're all on the same page.

File: envelope.md
Comment (trishankatdatadog) [inline envelope.md:5]: This is to account for a patch update that was not officially observed until now?

Comment (adityasaky) [reply envelope.md:5]: yes, we've had clarifications in the spec since v1.0.0 on `master` that was never tagged.

**Review by trishankatdatadog — APPROVED** [review]

**Review by lukpueh — APPROVED** [review]

**Comment by adityasaky** [comment]
Given we have three approvals and no comments expressing concerns, shall we merge and tag?

**Comment by MarkLodato** [comment]
SGTM

**Pull request merged**
